src/main/scala/edu/cmu/cs/obsidian/parser/Parser.scala

```diff
@@ -477,8 +492,15 @@ object Parser extends Parsers {
 
     private def parseContractDecl = {
         rep(parseContractModifier) ~ ContractT() ~! parseId ~!
-            LBraceT() ~! rep(parseDeclInContract) ~! RBraceT() ^^ {
-            case mod ~ ct ~ name ~ _ ~ defs ~ _ => Contract(mod.toSet, name._1, defs).setLoc(ct)
+            LBraceT() ~! rep(parseDeclInContract(false)) ~! RBraceT() ^^ {
```

Comment:
parseContractDecl and parseInterfaceDecl look remarkably similar. Can you avoid the code duplication?